### ISSUE TITLE
Native errors when running async-profiler with JNI library

### ISSUE THREAD

**Issue by barrett122** [async-profiler/async-profiler#933]
Hi folks,

While doing some performance evaluation of the [helm-java](https://github.com/manusa/helm-java) library we've come across a very unusual bug that we haven't been able to determine the cause of. Esentially, attempting to install helm releases with that library while async-profiler is running causes various errors from the native calls.
The most common seems to be a segmentation violation but there are a few others seemingly from Go(which helm is written in), null pointer dereferences, etc.
When the same program is run without async-profiler, or if the profiler was started before the JNI library has been loaded then it works normally, no errors of any kind and the expected releases get created.

I've put together a [minimal example repo](https://github.com/barrett122/helm-java-issue-demo) that reproduces the problem and I'll include a few of the error logs we've gotten from it below (though they're generally Go stack traces so I'm not sure how much help they'll be).

Given the nature of this issue we've found it super hard to track down which project this would relate to or what kind of solutions/workarounds we might be able to put in place. Any advice would be much appreciated!

[error.log](https://github.com/async-profiler/async-profiler/files/15348455/error.log)

[error2.log](https://github.com/async-profiler/async-profiler/files/15348458/error2.log)

[error3.log](https://github.com/async-profiler/async-profiler/files/15348462/error3.log)

[error4.log](https://github.com/async-profiler/async-profiler/files/15348466/error4.log)

([Corresponding helm-java issue](https://github.com/manusa/helm-java/issues/112))

**Comment by apangin** [comment]
Thank you for creating a reproducer. It is still not trivial to run, though. Is it possible to minimize the test case so that it does not require k8s setup?

Did you try running profiler with a different config, e.g., `ctimer` mode instead of `wall` and no `alloc` profiling? Does it still crash with other modes? `cstack` option might also make difference: try `cstack=dwarf` or `cstack=no`.

**Comment by barrett122** [comment]
Thanks for your response!

>It is still not trivial to run, though. Is it possible to minimize the test case so that it does not require k8s setup?

I had a look into this but unfortunately there isn't an obvious way around it. Even the helm `dryRun` mode still requires a working cluster 😞

>Did you try running profiler with a different config, e.g., ctimer mode instead of wall and no alloc profiling? Does it still crash with other modes? cstack option might also make difference: try cstack=dwarf or cstack=no.

So I tested all of the different events and cstack options, all of the `cpu`, `wall` and `ctimer` options cause it to fail, with any cstack option. `alloc` and `lock` both work but I wasn't able to test `cache-misses` due to some perf events error in the container that I wasn't able to debug easily (possibly a quirk of docker on mac, not sure)

**Comment by krk** [comment]
Thanks for sharing [the repro repo](https://github.com/barrett122/helm-java-issue-demo?tab=readme-ov-file#steps-to-run). I followed the steps provided and was unable to repro the crash, whether using the `true` or `false` argument (with or without async-profiler). Let me know if there are any additional details or configurations I should consider to replicate the issue more closely:

```
...
Installing: release46-flcoj
Installing: release49-mi6zp
Exception in thread "main" java.util.concurrent.TimeoutException: Futures timed out after [5 minutes]
```

I also noticed that the repo references `helm-java 0.0.7` in https://github.com/barrett122/helm-java-issue-demo/blob/8e74eb933d8d8e025e4d7354c0c11cf264645488/build.sbt#L13-L15, while version `0.0.11` has been released. Although its changelog doesn't mention an issue like this one on a cursory scan, I did see that the minimum golang version requirements changed.

Could you confirm if the issue can still be reproduced using the latest `helm-java` version (`0.0.11`) and the most recent golang compiler (`1.23.0` as of writing)?

**Comment by apangin** [comment]
Our initial investigation indicates the issue is not cause by async-profiler and may even occur without it.
I'm closing the issue in the lack of further details.
Feel free to reopen once you have an update.

**Comment by barrett122** [comment]
Hi folks,
Sorry for the delayed response, I missed the notifications for this. Thanks for taking a look!
So interestingly, I can still reproduce the issue just fine with the existing repo. I updated the helm-java version to the latest (`0.0.12`) and don't see any improvement unfortunately.
Tested this on macos and linux in case it was an OS thing, but issue seems the same across both.

It's interesting that you're seeing a timeout when running this @krk, generally when I run this either all the releases get created successfully with the profiler disabled, or I get a segmentation violation very quickly. I've not had a run that came close to hitting that 5 minutes so far, though that could all be down to external factors given the requests to the K8s cluster are probably the bottleneck here so maybe it's a red herring.
